README.md

```diff
@@ -0,0 +1,71 @@
+# http-wasm-abi
+
+This repository holds the [WebAssembly][1] Abstract Binary Interface (ABI) for
+HTTP middleware.
+
+## Overview
+
+WebAssembly has a virtual machine architecture where the *host* is the
+embedding process and the *guest* is a program compiled into the WebAssembly
+Binary Format, also known as Wasm. The Abstract Binary Interface (ABI) is the
+contract between the *host* and the *guest*, primarily defining functions each
+side can import.
+
+Let's take an example of a program that embeds an HTTP processor. This processor
+has a chain of middleware that allows it to customize or change requests and
+responses. Middleware in Wasm can be included in this chain.
+
+```goat
+ .----------------------. 
+|                        |
+|  .------------------.  |
+| | HTTP Processor     | |
+|  '------+-----------'  | 
+|         |  ^           |
+|         v  |           |
+|  .---------+--------.  |
+| | Native Middleware  | |
+|  '------+-----------'  | 
+|         |  ^           |
+|         v  |           |
+|  .---------+--------.  |
+| | Wasm Middleware    | |
+| |                    | |
+| | .-----.    .-----. | |
+| || Host  |  | Guest || |
+| ||       +->|       || |
+| ||       |<-+       || |
+| | '-----'    '-----' | |
+|  '------+-----------'  | 
+|         |  ^           |
+|         v  |           |
+|  .---------+--------.  |
+| | Native Middleware  | |
+|  '------------------'  | 
+|                        |
+ '----------------------' 
+```
+
+The Wasm Middleware *host* is written in native code, and compiled into the
+application. For example, this code could use [wazero][2] as the WebAssembly
+runtime, if the host was written in Go, or [V8][3], if C++.
+
+The Wasm Middleware *guest* can be written in any language that supports the
+ABI in use by the *host*. For example, the middleware could be programmed in C
```

Comment:
```suggestion
The Wasm Middleware *guest* can be written in any language that can be compiled
to Wasm. For example, the middleware could be programmed in C
```

I don't think there's any "support" a language provides for an ABI, it's the actual code that would use it.